SteamKit2/SteamKit2/Steam/CDNClient_Patch.cs

```diff
@@ -0,0 +1,101 @@
+        /// <param name="sourceManifestId">The unique identifier of the manifest to diff from.</param>
+        /// <param name="targetManifestId">The unique identifier of the manifest to diff to.</param>
+        /// <returns>A <see cref="DepotDeltaChunks"/> instance that contains information about the delta chunk patches present</returns>
+        public async Task<DepotDeltaChunks> DownloadPatchAsync( uint depotId, ulong sourceManifestId, ulong targetManifestId )
+        {
+            depotCdnAuthKeys.TryGetValue( depotId, out var cdnToken );
+            depotKeys.TryGetValue( depotId, out var depotKey );
+
+            return await DownloadPatchCoreAsync( depotId, sourceManifestId, targetManifestId, connectedServer, cdnToken, depotKey ).ConfigureAwait( false );
+        }
+
+        /// <summary>
+        /// Downloads the depot delta patch specified by the given manifest IDs.
+        /// </summary>
+        /// <param name="depotId">The id of the depot being accessed.</param>
+        /// <param name="sourceManifestId">The unique identifier of the manifest to diff from.</param>
+        /// <param name="targetManifestId">The unique identifier of the manifest to diff to.</param>
+        /// <param name="host">CDN hostname.</param>
+        /// <param name="cdnAuthToken">CDN auth token for CDN content server endpoints.</param>
+        /// <param name="depotKey">
+        /// The depot decryption key for the depot that will be downloaded.
+        /// This is used for decrypting filenames (if needed) in depot manifests, and processing depot chunks.
+        /// </param>
+        /// <returns>A <see cref="DepotDeltaChunks"/> instance that contains information about the delta chunk patches present</returns>
+        public async Task<DepotDeltaChunks> DownloadPatchAsync( uint depotId, ulong sourceManifestId, ulong targetManifestId, string host, string cdnAuthToken, byte[] depotKey )
+        {
+            var server = new Server
+            {
+                Protocol = Server.ConnectionProtocol.HTTP,
+                Host = host,
+                VHost = host,
+                Port = 80
+            };
+
+            return await DownloadPatchCoreAsync( depotId, sourceManifestId, targetManifestId, server, cdnAuthToken, depotKey ).ConfigureAwait( false );
+        }
+
+        async Task<DepotDeltaChunks> DownloadPatchCoreAsync( uint depotId, ulong sourceManifestId, ulong targetManifestId, Server server, string cdnAuthToken, byte[] depotKey )
+        {
+            if ( depotKey == null )
+            {
+                throw new ArgumentNullException( nameof( depotKey ) );
+            }
+
+            if ( sourceManifestId == targetManifestId )
+            {
+                throw new ArgumentException( "Source manifest cannot equal target manifest" );
+            }
+
+            // TODO: This request can return 404 with html content, need to account for that
```

Comment:
I haven't checked if the status code is actually 404, but the content is a `<html> [...] Not Found` error so the decryption fails on that currently.